### PR TITLE
Add Event Statistics Data Models and Repository

### LINKED ISSUE #288: Add Event Statistics Data Models and Repository
**Issue:** Add Event Statistics Data Models and Repository

Implement the data layer for the Event Statistics feature to enable organizers to view insights about their events. This includes data models for statistics (age distribution, campus distribution, join rate over time, attendees/followers rate), repository methods to fetch and compute these metrics, and the necessary string resources and chart colors.

### PR DESCRIPTION
**Pull request by rita-naimi**
> [!important] 
> Sonar low-importance warnings are linked to unused string resources that will be used in later PRs. 
They've been added here to not overload other PRs with too many lines of new code, since I had a hard time splitting the implementation of the Statistics Screen into PRs of reasonable size.

## What

This PR introduces the data layer foundation for the Event Statistics feature. It adds:
- Data models (`EventStatistics`, `AgeGroupData`, `CampusData`, `JoinRateData`) to represent event metrics
- `AgeGroups` utility object for age calculation and categorization
- `getEventStatistics()` method in `EventRepository` interface and its Firestore implementation
- Localized strings and chart color palette for future UI
- Unit tests with ~95% coverage

## Why

Organizations need insights into their events to understand their audience and measure engagement. This data layer enables fetching and computing key statistics:
- **Total attendees** - How many people joined
- **Age distribution** - Demographics breakdown (using same age ranges as org signup: <18, 18-22, 23-25, 26-30, 30+)
- **Campus distribution** - Which universities attendees come from
- **Join rate over time** - Registration pattern grouped by day
- **Attendees/followers rate** - Conversion metric showing what percentage of followers joined

## How

1. **Data Models** (`EventStatistics.kt`): Created immutable data classes with a companion `empty()` factory for initial/error states. `AgeGroups` provides utility functions to calculate age from DD/MM/YYYY birthdate strings and map ages to predefined groups.

2. **Repository Implementation** (`EventRepositoryFirestore.kt`): The `getEventStatistics()` method:
   - Fetches all event participants
   - For each participant, retrieves their user document to get birthday and university
   - Calculates age distribution using `AgeGroups` utilities
   - Calculates campus distribution (null → "Unknown")
   - Groups registrations by day for join rate timeline
   - Computes attendees/followers percentage (with zero-division guard)

3. **Resources**: Added 22 strings for statistics UI labels and 8 chart colors following a modern vibrant palette.

4. **Tests**: 9 unit tests covering all code paths including edge cases (empty participants, null data, Firestore exceptions, zero followers).

Fixes #288

### CODE REVIEW

File: app/src/main/java/com/github/se/studentconnect/model/event/EventRepositoryFirestore.kt [hunk 1]
```diff
@@ -416,4 +416,125 @@ class EventRepositoryFirestore(private val db: FirebaseFirestore) : EventReposit
...
+  override suspend fun getEventStatistics(eventUid: String, followerCount: Int): EventStatistics {
+    val participants = getEventParticipants(eventUid)
+    val totalAttendees = participants.size
+
+    // Fetch user data for participants to calculate distributions
+    val userDataList =
+        participants.mapNotNull { participant ->
+          try {
+            val userDoc = db.collection("users").document(participant.uid).get().await()
+            if (userDoc.exists()) {
```
Comment (othmanehousni) [inline app/src/main/java/com/github/se/studentconnect/model/event/EventRepositoryFirestore.kt]: For large events this performs one Firestore get() per participant, which can quickly become slow and expensive.
If possible, consider using a batched approach with whereIn on user IDs or some pre-aggregated data (even if we keep this version as a fallback). Or jut a TODO explaining the scalability limitation here would be helpful so we know this might need to be revisited.

File: app/src/main/java/com/github/se/studentconnect/model/event/EventStatistics.kt [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+      val currentMonth = now[java.util.Calendar.MONTH] + 1
+      val currentDay = now[java.util.Calendar.DAY_OF_MONTH]
+
+      var age = currentYear - year
+      if (currentMonth < month || (currentMonth == month && currentDay < day)) {
+        age--
+      }
+      age.takeIf { it >= 0 }
+    } catch (_: Exception) {
+      null
```
Comment (othmanehousni) [inline app/src/main/java/com/github/se/studentconnect/model/event/EventStatistics.kt]: This function hardcodes the birthday format as "DD/MM/YYYY". If the format used for user birthdays ever changes (or is locale-dependent elsewhere), this method will silently start producing wrong ages.

File: app/src/main/java/com/github/se/studentconnect/model/event/EventRepositoryFirestore.kt [hunk 3]
```diff
@@ -416,4 +416,125 @@ class EventRepositoryFirestore(private val db: FirebaseFirestore) : EventReposit
...
+                  count = count,
+                  percentage =
+                      if (totalAttendees > 0) (count.toFloat() / totalAttendees) * 100f else 0f)
+            }
+            .sortedByDescending { it.count }
+
+    // Calculate campus distribution
+    val campusGroups = mutableMapOf<String, Int>()
+    userDataList.forEach { (_, university, _) ->
+      val campus = university ?: "Unknown"
```
[the remark this answers is not in the document]
Comment (rita-naimi) [reply app/src/main/java/com/github/se/studentconnect/model/event/EventRepositoryFirestore.kt]: Good catch! But since this is the repository layer (no Context access), I'll use a constant instead. The UI layer can map it to the localized string. Do you agree with this way of doing it?
